### PR TITLE
[CAPI] add ov_get_last_error_msg() API

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
 - Add CAPI to get last error msg

### Tickets:
 - Closes https://github.com/openvinotoolkit/openvino/issues/20536

### CODE REVIEW

File: src/bindings/c/src/common.h [hunk 1]
```diff
@@ -224,3 +227,4 @@ struct mem_istream : virtual mem_stringbuf, std::istream {
 
 char* str_to_char_array(const std::string& str);
 ov::element::Type get_element_type(ov_element_type_e type);
+void dup_last_err_msg(const std::string msg);
```
Comment (zhaixuejun1993) [inline src/bindings/c/src/common.h:230]:
```suggestion
static inline void dup_last_err_msg(const std::string msg);
```

Comment (riverlijunjie) [reply src/bindings/c/src/common.h:230]: This function need access 2 shared variables cross multiple source code files, inline cannot work.

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/c/src/ov_core.cpp [hunk 2]
```diff
@@ -17,6 +17,24 @@ char* str_to_char_array(const std::string& str) {
     return char_array;
 }
 
+static char* last_err_msg = nullptr;
```
Comment (vurusovs) [inline src/bindings/c/src/ov_core.cpp:20]: is it possible to store `last_err_msg` as `std::string` and convert to `char*` directly in `ov_get_last_err_msg()`?

File: src/bindings/c/src/ov_core.cpp [hunk 3]
```diff
@@ -17,6 +17,24 @@ char* str_to_char_array(const std::string& str) {
...
+static std::mutex last_msg_mutex;
+void dup_last_err_msg(const std::string msg) {
```
Comment (vurusovs) [inline src/bindings/c/src/ov_core.cpp:22]: May be it was meant `dump_last_err_msg`?

Comment (riverlijunjie) [reply src/bindings/c/src/ov_core.cpp:22]: dup means duplicate.

File: src/bindings/c/src/ov_core.cpp (hunk 2 above)
Comment (riverlijunjie) [reply src/bindings/c/src/ov_core.cpp:20]: It seems the string contents will be lost if store exception.what() into std::string variable:
      `static std::string last_err_msg ;
       last_err_msg = std::string(ex.what())`

Maybe there is some mistake why the content has been lost in std::string.

Comment (riverlijunjie) [reply src/bindings/c/src/ov_core.cpp]: Fixed the content lost issue and use std::string, thank your suggestion! @vurusovs

**Review by vurusovs — APPROVED** [review]

**Pull request merged**